sed/io.py

```diff
@@ -1 +1,163 @@
-# TODO: populate
+import h5py
+import xarray as xr
+import numpy as np
+
+def recursive_write_metadata(h5group, node):
+    for key, item in node.items():
+        if isinstance(item, (np.ndarray, np.int64, np.float64, str, bytes, int, float, list)):
+            try:
+                h5group.create_dataset(key, data=item)
+            except TypeError:
```

Comment:
this is checked twice, here and in line 19

Reply:
They are different cases. Line 10 handles the case that one of the known data types cannot be saved as number. Line 19 tries to save everything unknown as string, and reports an error if also this fails. But maybe the logic could be changed.